packages/orchestrator-ui-components/src/pages/processes/timelineUtils.ts

```diff
@@ -0,0 +1,69 @@
+import { ProcessDetailStep, StepStatus } from '../../types';
+import { TimelineItem } from '../../components';
+
+export const getMostAccurateTimelineStatus = (
+    statusPreviousStep: StepStatus,
+    statusCurrentStep: StepStatus,
+): StepStatus =>
+    statusCurrentStep !== StepStatus.PENDING
+        ? statusCurrentStep
+        : statusPreviousStep;
+
+const mapStepToTimelineItem = (step: ProcessDetailStep): TimelineItem => {
+    return {
+        processStepStatus: step.status,
+        onClick: () => {
+            // Todo: Implement onClick after step-cards are implemented
+            // https://github.com/workfloworchestrator/orchestrator-ui/issues/225
+            console.log(`Clicked on ${step.name} (${step.stepid})`);
+        },
+    };
+};
+
+const stepsShouldBeMerged = (
+    previousStep: ProcessDetailStep,
+    currentStep: ProcessDetailStep,
+) =>
+    previousStep.status !== StepStatus.COMPLETE &&
```

Comment:
I would like to see a test for this logic. I think StepStatus.SUCCES should also be checked for the edge case you have 2 steps with the same name in succession.

Reply:
Good one, an probably SKIPPED as well

Reply:
Introduced a helper function to determine if a status is considered final